osu.Game/Overlays/Toolbar/ToolbarClock.cs

```diff
@@ -0,0 +1,323 @@
+                    };
+                }
+            }
+
+            private class SecondHand : CompositeDrawable
+            {
+                [BackgroundDependencyLoader]
+                private void load(OsuColour colours)
+                {
+                    RelativeSizeAxes = Axes.X;
+                    Width = 0.66f;
+
+                    Height = hand_thickness * 0.7f;
+                    Anchor = Anchor.Centre;
+                    Origin = Anchor.Custom;
+
+                    OriginPosition = new Vector2(Height * 2, Height / 2);
+
+                    InternalChildren = new Drawable[]
+                    {
+                        new Circle
+                        {
+                            Colour = colours.PinkLight,
+                            RelativeSizeAxes = Axes.Both,
+                        },
+                    };
+                }
+            }
+
+            private class LargeHand : CompositeDrawable
+            {
+                public LargeHand(float length)
+                {
+                    Width = length;
+                }
+
+                [BackgroundDependencyLoader]
+                private void load(OsuColour colours)
+                {
+                    Anchor = Anchor.Centre;
+                    Origin = Anchor.CentreLeft;
+
+                    Origin = Anchor.Custom;
+                    OriginPosition = new Vector2(hand_thickness / 2); // offset x also, to ensure the centre of the line is centered on the face.
+                    Height = hand_thickness;
+
+                    InternalChildren = new Drawable[]
+                    {
+                        new Circle
+                        {
+                            Colour = Color4.White,
+                            RelativeSizeAxes = Axes.Both,
+                            BorderThickness = 0.7f,
+                            BorderColour = colours.Gray2,
+                        },
+                    };
+
+                    RelativeSizeAxes = Axes.X;
+                }
+            }
+
+            protected override void UpdateDisplay(DateTimeOffset now)
+            {
+                float secondFractional = now.Second / 60f;
+                float minuteFractional = (now.Minute + secondFractional) / 60f;
+                float hourFractional = ((minuteFractional + now.Hour) % 12) / 12f;
+
+                updateRotation(hour, hourFractional);
+                updateRotation(minute, minuteFractional);
+                updateRotation(second, secondFractional);
+            }
+
+            private void updateRotation(Drawable hand, float fraction)
+            {
+                const float duration = 320;
+
+                float rotation = fraction * 360 - 90;
+
+                if (Math.Abs(hand.Rotation - rotation) > 180)
+                    hand.RotateTo(rotation);
+                else
+                    hand.RotateTo(rotation, duration, Easing.OutElastic);
+            }
```

Comment:
nit: Would be best to keep private nested classes at the bottom and move this above them. (I had a hard time trying to find how this works via diff until I checked via IDE)